### PR TITLE
Add sort selector to the Trusted Intermediary Group List page 

### LINKED ISSUE #7570: Alphabetize the list of TIs
#### Is your feature request related to a problem? Please describe.

We're getting a growing list of TIs and it can be the teenie tiniest bit challenging to find the right org since it is currently sorted chronologically by the date the organization was added.

#### Describe the solution you'd like.

Alphabetical sorting of the TI organization list, A - Z, based on the Name of the organization.

#### What priority is this?

Low, very low, but would be nice :)

#### Associated civic entity

Seattle

#### User Group

CiviForm Admin feedback

#### Additional Context

image
#### Done when:

What conditions need to be met before this issue can be closed?

### PR DESCRIPTION
**Pull request by jcma-google**
### Description

Add a sort selector to the TI groups page. Allow sorting by TI name asc, TI name desc, number of members desc and asc. 

- member asc 
image

- member desc
image

- name a-z
image

- name z-a
image

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary

### Issue(s) this completes

Fixes #7570

### CODE REVIEW

File: server/app/views/admin/ti/TrustedIntermediaryGroupListView.java
Comment (jcma-google) [inline server/app/views/admin/ti/TrustedIntermediaryGroupListView.java:85]: I still need to change QuestionSortOption to SortOption to reflect the more generic nature of the enums listed in the file. I am going to do so in my next PR, would prefer to keep this PR focused on adding the new sort selector.

File: browser-test/src/trusted_intermediary.test.ts
Comment (ssandbekkhaug) [inline browser-test/src/trusted_intermediary.test.ts:676]: Super nit: group aaa has 3 members

File: server/app/views/components/QuestionSortOption.java
Comment (ssandbekkhaug) [inline server/app/views/components/QuestionSortOption.java:9]: Just a thought: should there be multiple enums for different types of sorting, such as QuestionSortOption and TIGroupSortOption?

For instance, I think sorting questions by "TI name" would be an error, so the code shouldn't allow that.

**Comment by caitlinshk** [comment]
Hi, I (sadly) no longer work on CiviForm. @nb1701 can you help direct this?

File: server/app/views/components/QuestionSortOption.java
Comment (ssandbekkhaug) [reply server/app/views/components/QuestionSortOption.java:9]: I think putting QuestionSortOption and TiSortOptions as separate enums, as you suggested, is a good long term plan. In the short term, I'm OK with this PR. I will defer to the more experienced CiviForm developers, who may think otherwise. (I don't have a strong opinion on this.)

File: browser-test/src/trusted_intermediary.test.ts
Comment (ssandbekkhaug) [reply browser-test/src/trusted_intermediary.test.ts]: Is there supposed to be a code update? Or is there really only 2 members? (I see foo, foo2, and foo3)

**Review by ssandbekkhaug — DISMISSED** [review]

File: server/app/views/admin/ti/TrustedIntermediaryGroupListView.java
Comment (rockycodes) [inline server/app/views/admin/ti/TrustedIntermediaryGroupListView.java:159]: suggestion (blocking): since `cf-ti-info` is only used in tests, add it as a `data-testid` attribute. then you can use the `getByTestId` selector to select it directly. same for `cf-ti-member` below.

Example:
https://github.com/civiform/civiform/blob/6c96e1dd9ddd7a7b2f5f93dfc9d9620c1ac6dac5/server/app/views/applicant/TrustedIntermediaryClientListView.java#L182

https://github.com/civiform/civiform/blob/6c96e1dd9ddd7a7b2f5f93dfc9d9620c1ac6dac5/browser-test/src/trusted_intermediary.test.ts#L613

Also, double check if you should use `cf` for classes that are only used in tests. I think we generally use them as reference classes for js functionality, but tbh I'm not totally clear on the guidelines. We should talk about it as a group!

File: server/app/assets/javascripts/sort_selector.ts
Comment (rockycodes) [inline server/app/assets/javascripts/sort_selector.ts:6]: Is it weird that `tiname-asc` was already used as the example in this comment? This functionality is new, right? I'm wondering if I'm missing something.

File: browser-test/src/trusted_intermediary.test.ts
Comment (rockycodes) [inline browser-test/src/trusted_intermediary.test.ts:676]: suggestion (non-blocking): use `test.step` to avoid needing code comments and make it easier to debug failures: https://playwright.dev/docs/api/class-test#test-step

File: browser-test/src/trusted_intermediary.test.ts
Comment (rockycodes) [inline browser-test/src/trusted_intermediary.test.ts:686]: question (non-blocking): Did you intend to check for the existence of this group in particular?

**Review by rockycodes — CHANGES_REQUESTED** [review]
Overall I think this looks good! I agree with your plan of separating out the TI sort options. I mostly have comments about how you're using reference classes + a blocking comment about the name of your js file (I think you should make it more specific since we have several ti related pages).

Also, since you opted for the sort option/UI change, please run this by Sijia and/or Isabel to get their 👍 before merging if you haven't already.

File: browser-test/src/trusted_intermediary.test.ts
Comment (jcma-google) [reply browser-test/src/trusted_intermediary.test.ts:686]: No, this was left from a previous debugging effort.

File: server/app/views/admin/ti/TrustedIntermediaryGroupListView.java
Comment (jcma-google) [reply server/app/views/admin/ti/TrustedIntermediaryGroupListView.java:159]: i don't see the cf- prefix used for other test ids so I'll remove it

File: server/app/assets/javascripts/sort_selector.ts
Comment (jcma-google) [reply server/app/assets/javascripts/sort_selector.ts:6]: I extracted the method which was previously in question bank controller to this new shared method

**Comment by jcma-google** [comment]
isabel approved the UI changes!

File: server/app/assets/javascripts/admin_trusted_intermediary_list.ts
Comment (rockycodes) [inline server/app/assets/javascripts/admin_trusted_intermediary_list.ts:1]: Should this say:
```suggestion
/** The Trusted Intermediary controller is responsible for manipulating the trusted intermediary group list. */
```

**Review by rockycodes — APPROVED** [review]
lgtm!

**Pull request merged**
